### ISSUE TITLE
Worksheets not sorted when parsing an xls file

### ISSUE THREAD

**Issue by danipardo** [tafia/calamine#262]
If I open an xls format file, like in

```
    let workbook: Xls<_> = open_workbook("sample.xls").unwrap();
```
The vec returned by workbook.worksheets() seems to be in random order. I would expect them to be in order, otherwise we will have to rely on worksheet name to find the apropiate one. In my case, the user uploads an excel file, and I know the data is in the first worksheet, whatever name it has.

**Comment by cnpryer** [comment]
I'm interested in preserving order here too, but it looks like it's stored as a hashmap internally. So this is probably where order isn't preserved.

On a side note, I'm curious if this project is still maintained. 🤔

**Comment by Fuzen-py** [comment]
There has not been any updates from Tafia for quite some time. I sent the author an email. In the meantime, I will look into a solution for you, it may be adding a separate method ordered worksheets.

**Comment by cnpryer** [comment]
I'm open to contributing to someone's fork if there's enough support. Otherwise I'll probably just queue up PRs here and point at my own.

**Comment by Fuzen-py** [comment]
It looks like the worksheet's metadata is sorted, not really a great solution but may work for you?

```rust 
let workbook: Xls<_> = open_workbook("sample.xls").unwrap();
let mut sheets = workbook.worksheets();
let metadata = workbook.metadata();
let sheet_names = &metadata.sheets;

sheets.sort_by(|left, right| {
    let left_pos = sheet_names.iter().position(|v| left.0.eq(v));
    let right_pos = sheet_names.iter().position(|v| right.0.eq(v));

     left_pos.cmp(&right_pos)
});
```

**Comment by cnpryer** [comment]
I'll check it out. Thanks!

**Comment by Fuzen-py** [comment]
For @danipardo case, if you know the position `.sheet_names().get(2)` should give you the results you want as well if you fetch the range with `worksheet_range`. Again, this is under the assumption the metadata is in order. If not, we should probably store positioning there.

**Comment by danipardo** [comment]
Yes, workbook.sheet_names() seems to be sorted, so I guess we can work around with that. Thanks!

**Comment by danipardo** [comment]
Hi again, @Fuzen-py . Did you finally get a reply from @tafia about calamine project status? It's a pity the project is stuck, given how often people need to parse excel files.

**Comment by Fuzen-py** [comment]
@danipardo I have not heard back from the maintainer. I'm open to either forking this project or contributing to another one.

**Comment by danipardo** [comment]
@Fuzen-py I also sent an email to @tafia but didn't get reply either. Good news is that he merged two PR on July, which at least indicates he's still alive. I hope we hear from him soon!

**Comment by tafia** [comment]
Closing as fixed in #275
